packages/cojson/src/coValues/coMap.ts

```diff
@@ -39,44 +40,107 @@ export class RawCoMapView<
   /** @category 6. Meta */
   core: CoValueCore;
   /** @internal */
-  ops: {
+  latest: {
+    [Key in keyof Shape & string]?: MapOp<Key, Shape[Key]>;
+  };
+  /** @internal */
+  latestTxMadeAt: number;
+  /** @internal */
+  cachedOps?: {
     [Key in keyof Shape & string]?: MapOp<Key, Shape[Key]>[];
   };
   /** @internal */
+  options?: { ignorePrivateTransactions: boolean; atTime?: number };
+  /** @internal */
   atTimeFilter?: number = undefined;
   /** @category 6. Meta */
   readonly _shape!: Shape;
 
   /** @internal */
   constructor(
     core: CoValueCore,
-    options?: { ignorePrivateTransactions: true },
+    options?: { ignorePrivateTransactions: boolean; atTime?: number },
   ) {
     this.id = core.id as CoID<this>;
     this.core = core;
-    this.ops = {};
+    this.latest = {};
+    const latest = this.latest;
+    this.latestTxMadeAt = 0;
+    this.options = options;
+
+    const validTransactions = core.getValidSortedTransactions(options);
+
+    for (let i = validTransactions.length - 1; i >= 0; i--) {
+      const { txID, changes, madeAt } = validTransactions[i]!;
+
+      if (options?.atTime && madeAt > options.atTime) {
+        continue;
+      }
+
+      if (madeAt > this.latestTxMadeAt) {
+        this.latestTxMadeAt = madeAt;
+      }
+
+      for (let changeIdx = 0; changeIdx < changes.length; changeIdx++) {
+        const change = changes[changeIdx] as MapOpPayload<
+          keyof Shape & string,
+          Shape[keyof Shape & string]
+        >;
+        let entry = latest[change.key];
+        if (!entry) {
+          entry = {
+            txID,
+            madeAt,
+            changeIdx,
+            change,
+          };
+          latest[change.key] = entry;
+        } else if (madeAt > entry.madeAt) {
+          entry.txID = txID;
+          entry.madeAt = madeAt;
+          entry.changeIdx = changeIdx;
+          entry.change = change;
+        }
+      }
+    }
+  }
+
+  private getOps() {
+    if (this.cachedOps) {
```

Comment:
For the cases where we do need the full history, we get it lazily (and cache it)